addons/hr_holidays/static/src/js/time_off_calendar.js

```diff
@@ -117,34 +117,42 @@ odoo.define('hr_holidays.dashboard.view_custo', function(require) {
          *
          * @private
          */
-        _onNewAllocation: function () {
-            let self = this;
-
+        _onNewAllocation: async function () {
             let domain = [
                 ['name', '=', 'hr.leave.view.form.dashboard'],
                 ['model', '=', 'hr.leave.allocation']
             ];
 
-            self._rpc({
+            const ids = await this._rpc({
                 model: 'ir.ui.view',
                 method: 'search',
                 args: [domain],
-            }).then(function(ids) {
-                self.allocationDialog = new dialogs.FormViewDialog(self, {
-                    res_model: "hr.leave.allocation",
-                    view_id: ids,
-                    context: {
-                        'default_employee_ids': self.context.employee_id,
-                        'default_state': 'confirm',
-                    },
-                    title: _t("New Allocation"),
-                    disable_multiple_selection: true,
-                    on_saved: function() {
-                        self.reload();
-                    },
+            });
+
+            let default_employee_ids = this.context.employee_id;
+
+            if (!default_employee_ids) {
+                default_employee_ids = await this._rpc({
+                    model: 'hr.employee',
+                    method: 'search',
+                    args: [[['user_id', '=', this.context.uid]]]
```

Comment:
What happens in multi company contexts? This domain should contain the current company 🙂